pkg/build/builder/dockerutil.go

```diff
@@ -39,14 +44,18 @@ func pushImage(client DockerClient, name string, authConfig docker.AuthConfigura
 		opts.OutputStream = os.Stderr
 	}
 	var err error
+
 	for retries := 0; retries <= DefaultPushRetryCount; retries++ {
-		err = client.PushImage(opts, authConfig)
+		err := client.PushImage(opts, authConfig)
```

Comment:
yeah go doesn't do a very good job of warning you about shadowed variables.  for all the other stuff it disallows, i'm surprised it allows shadowing.